### PR TITLE
feat: AWS Lambda Powertools for Python templates

### PR DESCRIPTION
**Pull request by leandrodamascena**
*Issue #, if available:*
Issue number: #333 

*Description of changes:*
This example contains a cookiecutter template for creating a Python Hello world boilerplate using Serverless Application Model (SAM) and Powertools for Python features (https://awslabs.github.io/aws-lambda-powertools-python/latest).

Powertools is a suite of utilities for AWS Lambda functions to ease adopting best practices such as tracing, structured logging, custom metrics, idempotency, batching, and more.

In this example we create an API that has a GET /hello as an endpoint and that generates logs, metrics and traces using Powertools for Python.

Runtimes: python 3.7, python 3.8 and python 3.9

image
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: manifest-v2.json [hunk 1]
```diff
@@ -942,6 +942,14 @@
       "packageType": "Zip",
       "useCaseName": "Hello World Example"
     },
+    {
+      "directory": "python3.9/hello-pt",
```
Comment (jfuss) [inline manifest-v2.json:946]: This looks incomplete. PR has multiple Python version but the manifest is only updated for py3.9

**Review by jfuss — CHANGES_REQUESTED** [review]

File: manifest-v2.json (hunk 1 above)
Comment (leandrodamascena) [reply manifest-v2.json:962]: Hi @jfuss! Thanks for the review. Yep, was missing but I added it now.

File: python3.7/hello-pt/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **Python 3.7**: `sam init --runtime python3.9`
```
[the remark this answers is not in the document]
Comment (leandrodamascena) [reply python3.7/hello-pt/README.md]: Done

File: python3.8/hello-pt/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **Python 3.8**: `sam init --runtime python3.9`
```
[the remark this answers is not in the document]
Comment (leandrodamascena) [reply python3.8/hello-pt/README.md]: Done

File: tests/integration/unit_test/test_unit_test_python3_8.py [hunk 4]
```diff
@@ -5,6 +5,9 @@ class UnitTest_python3_8_cookiecutter_aws_sam_hello_python(UnitTestBase.Python38
     directory = "python3.8/hello"
     code_directories = ["hello_world"]
 
+class UnitTest_python3_9_cookiecutter_aws_sam_hello_pt_python(UnitTestBase.Python38UnitTestBase):
```
[the remark this answers is not in the document]
Comment (leandrodamascena) [reply tests/integration/unit_test/test_unit_test_python3_8.py]: Done

File: tests/integration/unit_test/test_unit_test_python3_7.py [hunk 5]
```diff
@@ -5,6 +5,9 @@ class UnitTest_python3_7_cookiecutter_aws_sam_hello_python(UnitTestBase.Python37
     directory = "python3.7/hello"
     code_directories = ["hello_world"]
 
+class UnitTest_python3_9_cookiecutter_aws_sam_hello_pt_python(UnitTestBase.Python37UnitTestBase):
```
Comment (jfuss) [inline tests/integration/unit_test/test_unit_test_python3_7.py:8]:
```suggestion
class UnitTest_python3_7_cookiecutter_aws_sam_hello_pt_python(UnitTestBase.Python37UnitTestBase):
```

**Review by jfuss — CHANGES_REQUESTED** [review]

**Review by jfuss — APPROVED** [review]
Thanks for handling the feedback

**Pull request merged**